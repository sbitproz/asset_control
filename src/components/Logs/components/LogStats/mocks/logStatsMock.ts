export const logsStatsMock = {
  INFO: 32100,
  WARNING: 33400,
  ERROR: 34500,
};
