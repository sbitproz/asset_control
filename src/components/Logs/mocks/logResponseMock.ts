export const logsResponseMock = {
  logs: [
    {
      dateTime: "2019-07-30 03:02:20,994",
      logType: "INFO",
      message:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
    },
    {
      dateTime: "2019-11-02 08:42:17,994",
      logType: "INFO",
      message: "Services",
    },
    {
      dateTime: "2019-11-30 10:15:23,994",
      logType: "ERROR",
      message: "Accounting",
    },
    {
      dateTime: "2019-09-03 22:24:28,994",
      logType: "INFO",
      message: "Sales",
    },
    {
      dateTime: "2020-06-08 04:23:35,994",
      logType: "INFO",
      message: "Human Resources",
    },
    {
      dateTime: "2020-05-02 13:19:02,994",
      logType: "WARNING",
      message: "Sales",
    },
    {
      dateTime: "2019-09-03 22:26:12,994",
      logType: "WARNING",
      message: "Human Resources",
    },
    {
      dateTime: "2020-05-31 06:07:54,994",
      logType: "INFO",
      message: "Accounting",
    },
    {
      dateTime: "2020-04-16 13:38:36,994",
      logType: "ERROR",
      message: "Sales",
    },
    {
      dateTime: "2020-04-12 19:04:36,994",
      logType: "INFO",
      message: "Product Management",
    },
    {
      dateTime: "2020-01-05 06:57:20,994",
      logType: "WARNING",
      message: "Product Management",
    },
    {
      dateTime: "2020-02-24 13:32:10,994",
      logType: "ERROR",
      message: "Research and Development",
    },
    {
      dateTime: "2020-05-27 12:37:20,994",
      logType: "WARNING",
      message: "Human Resources",
    },
    {
      dateTime: "2020-05-31 11:42:17,994",
      logType: "WARNING",
      message: "Research and Development",
    },
    {
      dateTime: "2019-08-17 04:15:32,994",
      logType: "INFO",
      message: "Product Management",
    },
    {
      dateTime: "2020-01-27 23:34:03,994",
      logType: "ERROR",
      message: "Services",
    },
    {
      dateTime: "2019-12-08 04:56:25,994",
      logType: "ERROR",
      message: "Human Resources",
    },
    {
      dateTime: "2020-02-05 19:03:52,994",
      logType: "ERROR",
      message: "Sales",
    },
    {
      dateTime: "2019-07-17 12:03:27,994",
      logType: "WARNING",
      message: "Research and Development",
    },
    {
      dateTime: "2019-11-18 21:28:01,994",
      logType: "WARNING",
      message: "Business Development",
    },
    {
      dateTime: "2019-09-11 22:55:38,994",
      logType: "ERROR",
      message: "Sales",
    },
    {
      dateTime: "2019-12-23 15:48:29,994",
      logType: "WARNING",
      message: "Accounting",
    },
    {
      dateTime: "2020-06-29 04:56:42,994",
      logType: "ERROR",
      message: "Business Development",
    },
    {
      dateTime: "2019-07-26 08:25:19,994",
      logType: "INFO",
      message: "Sales",
    },
    {
      dateTime: "2020-02-23 06:46:09,994",
      logType: "WARNING",
      message: "Engineering",
    },
  ],
  counter: {
    INFO: 32100,
    WARNING: 33400,
    ERROR: 34500,
  },
  pagination: {
    count: 25,
    offset: 1,
    limit: 25,
    total: 9999999,
  },
};

export const apiResponseMock = {
  data: logsResponseMock,
  status: 200,
  statusText: "COMPLETED",
  headers: undefined,
  config: {},
};
