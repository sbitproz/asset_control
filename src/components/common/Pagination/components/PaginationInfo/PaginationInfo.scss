@import "../../../../../styles/base/typography";

.pagination-info {
  max-width: 400px;
  font-weight: $font-weight-bold;
}
