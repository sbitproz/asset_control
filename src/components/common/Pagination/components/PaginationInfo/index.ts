export { default as PaginationInfo } from "./PaginationInfo";
