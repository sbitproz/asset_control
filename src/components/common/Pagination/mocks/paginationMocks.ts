export const pagniationInfoMocks = {
  count: 25,
  offset: 1,
  limit: 25,
  total: 9999999,
};
