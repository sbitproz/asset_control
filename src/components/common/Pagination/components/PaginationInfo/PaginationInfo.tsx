import React, { memo } from "react";

// Own
import { PaginationProperties } from "../../interfaces/paginationInterfaces";

// Styles
import "./PaginationInfo.scss";

const PaginationInfo: React.SFC<PaginationProperties> = ({
  pageInformation,
}) => {
  const getPosition = (): number =>
    pageInformation.offset + pageInformation.count === pageInformation.total
      ? 100
      : (pageInformation.offset / pageInformation.total) * 100;

  return (
    <div data-testid="pagination-info" className="pagination-info">
      {pageInformation.total === 0
        ? "No Logs"
        : `Show ${pageInformation.offset + 1} to ${
            pageInformation.count + pageInformation.offset
          }
      of ${pageInformation.total} (${getPosition().toFixed(2)}%)`}
    </div>
  );
};

export default memo(PaginationInfo);
