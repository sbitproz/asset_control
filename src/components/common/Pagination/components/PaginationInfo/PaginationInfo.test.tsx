import React from "react";
import { shallow } from "enzyme";

// Own
import PaginationInfo from "./PaginationInfo";
import { pagniationInfoMocks } from "../../mocks/paginationMocks";
import { testSelector } from "../../../../../tests/testHelper";

describe("PaginationInfo", () => {
  let wrapper: any;

  const noLogs = "No Logs";
  const withLogs = "Show";

  const percentageWith1Record = ((1 / pagniationInfoMocks.total) * 100).toFixed(
    2
  );
  const percentageWith999999Record = (
    (999999 / pagniationInfoMocks.total) *
    100
  ).toFixed(2);

  beforeEach(() => {
    wrapper = shallow(<PaginationInfo pageInformation={pagniationInfoMocks} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should show pagination summary details: end range (count + offset)", () => {
    expect(wrapper.find(`${testSelector("pagination-info")}`).text()).toContain(
      `${pagniationInfoMocks.offset + pagniationInfoMocks.count}`
    );
  });

  it("should show pagination summary details: offset", () => {
    expect(wrapper.find(`${testSelector("pagination-info")}`).text()).toContain(
      `${pagniationInfoMocks.offset + 1}`
    );
  });

  it("should show pagination summary details: total", () => {
    expect(wrapper.find(`${testSelector("pagination-info")}`).text()).toContain(
      `${pagniationInfoMocks.total}`
    );
  });

  it(`should show 'Show' when total is greater than 0`, () => {
    expect(wrapper.find(`${testSelector("pagination-info")}`).text()).toContain(
      withLogs
    );
  });

  it(`should show 'no logs' when total is 0`, () => {
    wrapper.setProps({ pageInformation: { ...pagniationInfoMocks, total: 0 } });
    expect(wrapper.find(`${testSelector("pagination-info")}`).text()).toContain(
      noLogs
    );
  });

  it(`should show 'no logs' when total is 0`, () => {
    wrapper.setProps({ pageInformation: { ...pagniationInfoMocks, total: 0 } });
    expect(wrapper.find(`${testSelector("pagination-info")}`).text()).toContain(
      noLogs
    );
  });

  it(`should show ${percentageWith1Record} when there's 1 record`, () => {
    wrapper.setProps({
      pageInformation: { ...pagniationInfoMocks, offset: 1 },
    });
    expect(wrapper.find(`${testSelector("pagination-info")}`).text()).toContain(
      percentageWith1Record
    );
  });

  it(`should show ${percentageWith999999Record} when there's 2 records`, () => {
    wrapper.setProps({
      pageInformation: { ...pagniationInfoMocks, offset: 999999 },
    });
    expect(wrapper.find(`${testSelector("pagination-info")}`).text()).toContain(
      percentageWith999999Record
    );
  });
});
