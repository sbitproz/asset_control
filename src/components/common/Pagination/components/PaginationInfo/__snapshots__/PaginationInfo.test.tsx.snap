// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationInfo should match snapshot 1`] = `
<div
  className="pagination-info"
  data-testid="pagination-info"
>
  Show 2 to 26
      of 9999999 (0.00%)
</div>
`;
